test-tsd/batchInsert.test-d.ts

```diff
@@ -0,0 +1,47 @@
+import Knex from '../types';
+import { expectError, expectType } from 'tsd';
+import { clientConfig } from './common';
+
+const knex = Knex(clientConfig);
+
+// Use:
+// import Knex from 'knex'
+// when "esModuleInterop": true
+
+const main = async () => {
+  // default returning is number[]
+  expectType<number[]>(await knex.batchInsert('table', [{a: 2, id: 3}] as const, 12));
+
+  // allows any data
+  expectType<any[]>(await knex.batchInsert('table', [{a: 2, id: 3}] as any[], 12).returning('abc'));
+
+  // allows any data
+  expectType<any[]>(await knex.batchInsert('table', [{a: 2, id: 3}] as any, 12).returning('abc'));
+
+  // returning '*' allowed, return full record
+  expectType<Array<{a: boolean, id: number}>>(await knex.batchInsert('table', [{a: false, id: 3}], 12).returning('*'));
+
+  // one column for returning
+  expectType<Array<{aa2: string}>>(await knex.batchInsert('table', [{aa2: 'string', b: 3}], 12).returning('aa2'));
+
+  // many columns for returning
+  expectType<Array<{aa2: string, b: boolean}>>(await knex.batchInsert('table', [{aa2: 'string', b: false, c: 12}], 12).returning(['aa2', 'b']));
+
+  // assert bad column in returning gives error
+  expectError(
+    knex.batchInsert('table', [{aa2: 'string', b: false, c: 12}], 12)
+      .returning('bad_column')
+  );
+
+  // assert any bad column from returning list gives error
+  // tsd gives error on this, lol
+  /*expectError(
```

Comment:
Do you think that tsd is wrong here? If there is an easy self-contained repro possible, perhaps an issue at https://github.com/SamVerschueren/tsd/issues could be opened?

Reply:
https://github.com/SamVerschueren/tsd/issues/87

Reply:
Thank you. Can you add the reference to that bug report next to the commented out code?

Reply:
I will try to fix it fast in tsd

Reply:
does tsd have any caches? I cant reproduce it anymore on extracted example

Reply:
Not tsd per se, but TypeScript definitely has some caches, so that might have been a factor.

Reply:
Oh, i copied stacktrace from wrong place

Reply:
Error pops up because this expression give 2 errors, but expectError allows only one

Reply:
alternative fix is expectError(expectError(<expression>))